components/calendar/src/buddhist.rs

```diff
@@ -0,0 +1,139 @@
+        Iso.days_in_year(&date.0)
+    }
+
+    fn days_in_month(&self, date: &Self::DateInner) -> u8 {
+        Iso.days_in_month(&date.0)
+    }
+
+    fn offset_date(&self, date: &mut Self::DateInner, offset: DateDuration<Self>) {
+        Iso.offset_date(&mut date.0, offset.cast_unit())
+    }
+
+    #[allow(clippy::field_reassign_with_default)] // it's more clear this way
+    fn until(
+        &self,
+        date1: &Self::DateInner,
+        date2: &Self::DateInner,
+        largest_unit: DateDurationUnit,
+        smallest_unit: DateDurationUnit,
+    ) -> DateDuration<Self> {
+        Iso.until(&date1.0, &date2.0, largest_unit, smallest_unit)
+            .cast_unit()
+    }
+
+    /// The calendar-specific year represented by `date`
+    fn year(&self, date: &Self::DateInner) -> types::Year {
+        iso_year_as_buddhist(date.0.year)
+    }
+
+    /// The calendar-specific month represented by `date`
+    fn month(&self, date: &Self::DateInner) -> types::Month {
+        Iso.month(&date.0)
+    }
+
+    /// The calendar-specific day-of-month represented by `date`
+    fn day_of_month(&self, date: &Self::DateInner) -> types::DayOfMonth {
+        Iso.day_of_month(&date.0)
+    }
+
+    /// Information of the day of the year
+    fn day_of_year_info(&self, date: &Self::DateInner) -> types::DayOfYearInfo {
+        let prev_year = IsoYear(date.0.year.0 - 1);
+        let next_year = IsoYear(date.0.year.0 + 1);
+        types::DayOfYearInfo {
+            day_of_year: Iso::day_of_year(date.0),
+            days_in_year: Iso::days_in_year(date.0.year),
+            prev_year: iso_year_as_buddhist(prev_year),
+            days_in_prev_year: Iso::days_in_year(prev_year),
+            next_year: iso_year_as_buddhist(next_year),
+        }
+    }
+
+    fn debug_name() -> &'static str {
+        "Buddhist"
+    }
+}
+
+impl Date<Buddhist> {
+    /// Construct a new Buddhist Date
+    pub fn new_buddhist_date(
+        year: IsoYear,
+        month: IsoMonth,
+        day: IsoDay,
+    ) -> Result<Date<Buddhist>, DateTimeError> {
+        Date::new_iso_date(year, month, day).map(|d| Date::new_from_iso(d, Buddhist))
+    }
+}
+
+impl DateTime<Buddhist> {
+    /// Construct a new Buddhist datetime from integers
+    ///
+    /// Years are specified as BE years
+    pub fn new_buddhist_datetime_from_integers(
+        year: i32,
+        month: u8,
+        day: u8,
+        hour: u8,
+        minute: u8,
+        second: u8,
+    ) -> Result<DateTime<Buddhist>, DateTimeError> {
+        let iso_year = year - BUDDHIST_ERA_OFFSET;
+        Ok(DateTime {
+            date: Date::new_buddhist_date(iso_year.into(), month.try_into()?, day.try_into()?)?,
+            time: types::Time::try_new(hour, minute, second)?,
+        })
+    }
+}
+
+fn iso_year_as_buddhist(year: IsoYear) -> types::Year {
+    let buddhist_year = year.0 + BUDDHIST_ERA_OFFSET;
+    types::Year {
+        era: types::Era(tinystr8!("be")),
+        number: buddhist_year,
```

Comment:
Question: this creates negative years in the "be" era.  Is that correct, or is there a BCE-like era in this calendar?

Reply:
It's "correct" in that the calendar is not supposed to be used for past dates, but we already decided that we won't support cutoff dates for calendars and instead will fall back to negative eras.

Reply:
That's right; I was asking if this calendar system has an era for old dates, something like "BBE" (before buddhist era)?  If not, then negative dates in the current era is fine.